### PR TITLE
lightning: revert custom gRPC resolver

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/33114

Problem Summary:
https://github.com/pingcap/tidb/pull/32876 introduces a custom gRPC resolver to  tikv node address changes. However, it doesn't use tls config correctly. 

### What is changed and how it works?

Partial revert the custom gRPC part of https://github.com/pingcap/tidb/pull/32876. 
Custom gRPC is not needed. See https://github.com/pingcap/tidb/issues/32875#issuecomment-1068693681.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/split_client.go [hunk 1]
```diff
@@ -64,7 +63,7 @@ type SplitClient interface {
 	ScatterRegions(ctx context.Context, regionInfo []*RegionInfo) error
 	// GetOperator gets the status of operator of the specified region.
 	GetOperator(ctx context.Context, regionID uint64) (*pdpb.GetOperatorResponse, error)
-	// ScanRegions gets a list of regions, starts from the region that contains key.
+	// ScanRegion gets a list of regions, starts from the region that contains key.
```
Comment (glorv) [inline br/pkg/restore/split_client.go:66]: 🤔

Comment (gozssky) [reply br/pkg/restore/split_client.go:66]: https://github.com/pingcap/tidb/pull/32876 fixed the typo, but this PR revert it.

Comment (gozssky) [reply br/pkg/restore/split_client.go]: I have fixed it again.

**Review by glorv — COMMENTED** [review]
rest LGTM

File: br/tests/lightning_fail_fast/run.sh [hunk 2]
```diff
@@ -27,8 +27,7 @@ for CFG in chunk engine; do
 
   tail -n 10 $TEST_DIR/lightning-tidb.log | grep "ERROR" | tail -n 1 | grep -Fq "Error 1062: Duplicate entry '1-1' for key 'uq'"
 
-  ! grep -Fq "restore file completed" $TEST_DIR/lightning-tidb.log
-  [ $? -eq 0 ]
+  grep -Fq "restore file failed" $TEST_DIR/lightning-tidb.log
```
[the remark this answers is not in the document]
Comment (gozssky) [reply br/tests/lightning_fail_fast/run.sh]: Fixed.

**Review by glorv — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**